website/static/js/addProjectPlugin.js

```diff
@@ -0,0 +1,178 @@
+                            'category': self.newProjectCategory,
+                            'description' : self.newProjectDesc()
+                        }
+                    }
+                };
+            var success = function _success (result) {
+                self.viewState('success');
+                console.log('success', result);
+                self.goToProjectLink(result.data.links.html);
+            };
+            var error = function _error (result) {
+                self.viewState('error');
+                console.log('error', result);
+            };
+            m.request({method : 'POST', url : url, data : data, config : xhrconfig})
+                .then(success, error);
+            self.newProjectName('');
+            m.redraw(true);
+
+        };
+        self.reset = function _reset(){
+            self.newProjectName('');
+            self.viewState('form');
+            $('.modal').modal('hide');
+        };
+    },
+    view : function (ctrl, options) {
+        var templates = {
+            form : m('.modal-content', [
+                m('.modal-header', [
+                    m('button.close[data-dismiss="modal"][aria-label="Close"]',{ onclick : ctrl.reset}, [
+                        m('span[aria-hidden="true"]','×'),
+                    ]),
+                    m('h3.modal-title#addProject', 'Add New Project')
+                ]),
+                m('.modal-body', [
+                    m('', [
+                        m('.form-group.m-v-sm', [
+                            m('label[for="projectName].f-w-lg.text-bigger', 'Project Name'),
+                            m('input[type="text"].form-control#projectName', {
+                                onkeyup: function(ev){
+                                    if (ev.which === 13) {
+                                         ctrl.add();
+                                    }
+                                    ctrl.newProjectName($(this).val());
+                                },
+                                value : ctrl.newProjectName()
+                            })
+                        ]),
+                        m('.form-group.m-v-sm', [
+                            m('label[for="projectDesc].f-w-lg.text-bigger', 'Project Description'),
+                            m('textarea.form-control#projectDesc', {
+                                onkeyup: function(ev){
+                                    ctrl.newProjectDesc($(this).val());
+                                },
+                                value : ctrl.newProjectDesc()
+                            })
+                        ]),
+                        m('.f-w-lg.text-bigger','Category'),
+                        m('.category-radio.p-h-md', [
+                            m('.radio', m('label', [m('input[type="radio"][name="projectCategory][id="catProject][value="project][checked="checked"]',{ onchange : ctrl.chooseCategory }), 'Project'])),
```

Comment:
Good catch, I was trying to avoid using m.withatt but with a bit more work I got that to work and removed repitition in the code with this:
```js
m('.category-radio.p-h-md', [
    ctrl.categoryList.map(function(cat){
        return m('.radio', m('label', [  m('input', {
            type: 'radio',
            name: 'projectCategory',
            id: cat.id,
            value: cat.value,
            checked: ctrl.newProjectCategory() === cat.value,
            onchange : m.withAttr('value', ctrl.newProjectCategory)
        }), cat.label ]));
    })
])
```